src/main/java/Aly.java

```diff
@@ -0,0 +1,242 @@
+        printLine();
+    }
+
+    private static int addTodo(Task[] listItems, String task, int index) {
+        printLine();
+        listItems[index] = new Todo(task.trim());
+        index += 1;
+        System.out.println("Added this task: " + task.trim());
+        System.out.println("You have " + Task.getTaskCounter() + " tasks in your list now.");
+        printLine();
+        return index;
+    }
+
+    private static int addDeadline(Task[] listItems, String task, int index) {
+        printLine();
+        String[] taskParts = task.split("/by");
+        if (taskParts.length != 2) {
+            System.out.println("Wrong format!");
+            printLine();
+            return index;
+        }
+        String taskDeadline = taskParts[0].trim();
+        String taskBy = taskParts[1].trim();
+        listItems[index] = new Deadline(taskDeadline, taskBy);
+        index += 1;
+        System.out.println("Added this task: " + taskDeadline);
+        System.out.println("You have " + Task.getTaskCounter() + " tasks in your list now.");
+        printLine();
+        return index;
+    }
+
+    private static int addEvent(Task[] listItems, String task, int index) {
+        printLine();
+        String[] taskParts = task.split("/from|/to");
+        if (taskParts.length != 3) {
+            System.out.println("Wrong format!");
+            printLine();
+            return index;
+        }
+        String taskEvent = taskParts[0].trim();
+        String taskFrom = taskParts[1].trim();
+        String taskTo = taskParts[2].trim();
+        listItems[index] = new Event(taskEvent, taskFrom, taskTo);
+        index += 1;
+        System.out.println("Added this task: " + taskEvent);
+        System.out.println("You have " + Task.getTaskCounter() + " tasks in your list now.");
+        printLine();
+        return index;
+    }
+
+    private static void markTask(Task[] taskList) {
+        boolean isExit = false;
+        int index = 0;
+        index = getIndex(taskList, index);
+        while (!isExit) {
+            System.out.println("Enter 'list' to see task list, 'mark'/'unmark' with a number to toggle respective task status, '0' to exit");
+            String input = in.nextLine().trim();
+            String[] splitInput = input.split(" ");
+            String firstWord = splitInput[0];
+
+            switch (firstWord) {
+            case "list":
+                listTasks(taskList);
+                break;
+            case "mark":
+            case "unmark":
+                handleMarking(firstWord, splitInput, input, index, taskList);
+                break;
+            case "0":
+                isExit = true;
+                System.out.println("Returning to main menu!");
+                break;
+            default:
+                printLine();
+                System.out.println("Can read instructions anot? Try again!");
+                printLine();
+                break;
+            }
+        }
+    }
```

Comment:
There appears to be quite a bit of overlap between this "markTask" method and the above "list" method, perhaps they could be rewritten together to reduce duplicated code?